docs/blog/2021-06-21-zephyr-2-5.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Zephyr 2.5 Update"
+author: Pete Johanson
+author_title: Project Creator
+author_url: https://gitlab.com/petejohanson
+author_image_url: https://www.gravatar.com/avatar/2001ceff7e9dc753cf96fcb2e6f41110
+tags: [firmware, zephyr, core]
+---
+
+Happy to announce that we have completed the work to upgrade ZMK to [Zephyr 2.5](https://docs.zephyrproject.org/2.5.0/releases/release-notes-2.5.html)!
+
+A big part of this work was some _major_ refactors and improvements by [innovaker] to our [zmk-docker](https://github.com/zmkfirmware/zmk-docker/) Docker image and GH Actions automation. With the updated changes there, we can now:
+
+- GH and layer caching for improved build times.
+- Automated tests run against newly built images to verify them.
+- PRs to the repo now properly build and run the tests as well.
+- Build images for multiple target architectures, e.g. `zmk-build-riscv64`, all in parallel.
+- Create nightly builds to be sure we're pulling in the latest OS/package updates, to ensure we keep our images up to date, address any reported vulnerabilities, etc.
+- Allow us to upgrade to newer Zephyr SDK and Zephyr versions, more quickly.
+- Publish candidate tags, and then promote tested ones to final tags like `zmk-dev-arm:2.5`.
+
+## Getting The Changes
+
+Use the following steps to update to the latest tooling in order to properly use the new ZMK changes:
+
+### Github Actions
+
+Existing Github Actions will pull down Zephyr 2.5 automatically, and should work, fine as is. However, to be safe, you should edit the `.github/workflows/build.yml` to change `zmkfirmware/zmk-build-arm:2.4` to `zmkfirmware/zmk-build-arm:2.5`.
+
+### VSCode Docker Integration
+
+If you build locally using the VSCode Docker integration, then after pulling the latest ZMK `main` and reloading the project, you should be prompted to rebuild the remote container, Click Rebuild and once completed, VSCode will be running using the updated Docker image.
+
+If you're not prompted to reload, hit `Ctrl` + `Shift` + `P` and then run the `Remote Containers: Rebuild Container`.
+
+### Local Toolchain
+
+Your existing Zephyr SDK should build ZMK properly, but if you want to update to the latest, grab the [0.12.4 release](https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.12.4) and follow the instructions to install it.
+
+## Getting The Code
+
+If building locally or in VSCode with Docker, simply pull the latest changes from `main` to get the ZMK updates to move to Zephyr 2.5.
+
+## West Module Updates
+
+Updated Zephyr needs updated Zephyr dependencies as well. Run the following in your ZMK checkout:
+
+```
+west update
+```
+
+From there, you should be ready to build as normal!
+
```

Comment:
```suggestion
```

It's not completely clear (visually) that:
- `West Module Updates` is a separate section to `Getting The Changes`
- it must be done afterwards
  - but *not* for GitHub Actions.

I can see users falling into #help because they didn't `west update`.

Perhaps add it individually into each relevant sub-section of `Getting The Changes` (perhaps call it `Getting The Update`).

Reply:
Yeah, I moved this into the VS Code/Docker and Local Host Development sections, and removed it as a separate section.